main.rb

```diff
@@ -54,38 +54,30 @@
 end
 
 module Memo
-  JSON_FILE_NAME = 'memo_data.json'
+  TABLE = 'memo'
+  DB_CONNECTION = PG.connect(host: 'localhost', port: 5432, dbname: 'memo_app', user: 'memoapp_user')
 
   class << self
     def all
-      memo_data = JSON.parse(File.read(JSON_FILE_NAME), symbolize_names: true)
-      memo_data[:memo_list]
+      memo_data = DB_CONNECTION.exec("SELECT * FROM #{TABLE}")
+      memo_data.map { |memo| memo.transform_keys(&:to_sym) }
     end
 
     def find(target_id)
-      memo_data = JSON.parse(File.read(JSON_FILE_NAME), symbolize_names: true)
-      memo_data[:memo_list].find { |memo| memo[:id] == target_id }
+      result = DB_CONNECTION.exec("SELECT * FROM #{TABLE} WHERE id = $1::int", [target_id])
```

Comment:
`$1` が `int` に typecast されているのはなにか理由があるんでしょうか？ 🤔 また `Memo.all` のほうでは結果が代入される変数は `memo_data` になっていますが、こちらでは `result` となっていて名前に一貫性がないように思いました。一貫性のある名前をつけるのがよいでしょう。

Reply:
- $1 の int 指定に関して

こちらの[RubyGemのドキュメント](https://www.rubydoc.info/gems/pg/PG/Connection#exec_params-instance_method)に

> If the types are not specified, they will be inferred by PostgreSQL. Instead of specifying type oids, it's recommended to simply add explicit casts in the query to ensure that the right type is used.<br>
For example: “SELECT $1::int”

と表記されており、こちらの[PostgreSQLのドキュメント](https://www.postgresql.jp/document/9.4/html/datatype-numeric.html)にserial型が整数列を作成すると記載されているので`int`で明示しました！

ただ同記事内でserial型はデータ型ではなく識別子とも説明されているので削除したほうが良いでしょうか？

<br>

- 変数名の一貫性に関して

戻り値を受け取る変数名は単数なら`memo`、複数なら`memos`として一貫性があるように修正しました！

https://github.com/ymmtd0x0b/sinatra-memoapp/blob/843d8522776ba6d9abf3d5723417e2895e08828c/main.rb#L74-L75

https://github.com/ymmtd0x0b/sinatra-memoapp/blob/843d8522776ba6d9abf3d5723417e2895e08828c/main.rb#L79-L80

Reply:
> と表記されており、こちらの[PostgreSQLのドキュメント](https://www.postgresql.jp/document/9.4/html/datatype-numeric.html)にserial型が整数列を作成すると記載されているのでintで明示しました！

あーなるほど。であればこちらでよいと思います！ 🙆